### DISCUSSION TITLE
Ocarina of Time Randomizer Save flag issues

### DISCUSSION THREAD

**Discussion by Millennium-Eagle** [mupen64plus-ae/mupen64plus-ae#945] [General]
A while ago, I had an issue in an Ocarina of Time Triforce Hunt Rando where loading an autosave state could cause me to lose triforce pieces; permanantly if I saved the game afterwards, as the chests that contained them would stay open. After that, I turned off autosave states, but I have since learned that this isn't an issue that should necessarily happen with all emulators, and in fact, does not seem to happen with most of them.

More importantly, however, after playing tons of Randos with no issues, I discovered that one of the skulltulas I hadn't collected yet was missing (the one on the graveyard wall as child). I know I didn't forget about collecting it because it required boomerang and I had only just gotten boomerang. I might have killed it with slingshot to see what it was, but I don't remember, and killing it shouldn't remove it unless you collect the item it drops. Luckily, it was only a couple of rupees, according to the spoiler log, but if it was anything important I would have been screwed. I reloaded the area and changed time of day multiple times to see if it would reappear, but to no avail.

I reported the issue to the devs in the Oot Rando discord, but they seemed to be under the impression that it was an emulator issue, as they haven't heard of an issue like this happening before in one of their officially supported emulators. I also brought up the Triforce Hunt issue, and they also haven't heard of that problem before, which more or less confirmed it was an emulator issue for them.

I don't know what could cause this issue, but it might also be worth nothing that I have never, ever encountered a TLB exception in this emulator, despite this emulator having the same workaround for it in its options as Retroarch's Mupen64 Plus Next does. I don't know if that helps work out any of the differences between this emulator and other Mupen64 Plus versions that could cause these other issues.

**Comment by fzurita** [comment]
Interesting, this should be identical to the RetroArch version. As to how it may have happened, I truly have no idea, it would have been a problem with the core itself, if it's not a game bug. I don't do anything special with the core that could cause something like this.

If it really is a problem, it would be good to come up with a reproducible test case so that it can be figured out.

**Comment by flagrama** [comment]
Do you change the file names of the ROMs before adding them to the phone? And/or do you have `Settings > Data > Use flat game data folder structure` enabled?

**Reply by Millennium-Eagle** [reply]
No, The randomizer roms have the same filenames they were generated with. I have never enabled that setting, either.

**Reply by fzurita** [reply]
Neither of those settings should really cause in-game bugs.

**Reply by flagrama** [reply]
I admit it probably shouldn't affect autosave/load because that goes into the game-specific folder anyway from what I saw, but renaming a randomizer ROM to the same name as you have used for another one with using the flat folder structure and opening the same save slot in the new ROM will 100% cause a bug in randomizer if starting a ROM from boot. Every individual randomizer ROM sets different flags on file creation and using a save from one ROM on another is 100% incompatible.

I don't think anyone who plays randomizer uses auto save/load on RetroArch though so the same bug could be there somehow and nobody has ever run into because it is not default, we do not tell people how to activate it, and if someone were using it, it'd probably be the first thing we told them to disable tbh. We've had so many issues with the way emulators save the files that anything that does something weird with the saves is usually the first thing we tell them to change.
